src/main/resources/static/js/multitool/PdfContainer.js

```diff
@@ -280,28 +278,71 @@ class PdfContainer {
       this.fileName = filenameInput.value;
     }
 
-    if (!filenameInput.value.includes(".pdf")) {
-      filenameInput.value = filenameInput.value + ".pdf";
-      this.fileName = filenameInput.value;
-    }
+    const separators = this.pagesContainer.querySelectorAll(".split-before");
+    if (separators.length !== 0) { // Send the created blob to the split-pages API endpoint if there are separators.
+      const formData = new FormData();
 
-    if (downloadOption === "sameWindow") {
-      // Open the file in the same window
-      window.location.href = url;
-    } else if (downloadOption === "newWindow") {
-      // Open the file in a new window
-      window.open(url, "_blank");
-    } else {
-      // Download the file
-      this.downloadLink = document.createElement("a");
-      this.downloadLink.id = "download-link";
-      this.downloadLink.href = url;
-      // downloadLink.download = this.fileName ? this.fileName : 'managed.pdf';
-      // downloadLink.download = this.fileName;
-      this.downloadLink.setAttribute("download", this.fileName ? this.fileName : "managed.pdf");
-      this.downloadLink.setAttribute("target", "_blank");
-      this.downloadLink.onclick = this.setDownloadAttribute;
-      this.downloadLink.click();
+      const pagesArray = Array.from(this.pagesContainer.children);
+
+      let splitters = "";
+      separators.forEach(page => {
+        const pageIndex = pagesArray.indexOf(page);
+        if (pageIndex !== 0) {
+          splitters += pageIndex + ",";
+        }
+      });
+
+      formData.append("pageNumbers", splitters);
+      formData.append("fileInput", pdfBlob, this.fileName ? this.fileName : "managed.pdf");
+
+      const response = await fetch("/api/v1/general/split-pages", {
```

Comment:
ideally, do with the client side to reduce server load etc

Reply:
Gotcha, will have a look at that soon!

Reply:
Done! Requesting another review.